docs/src/ci.md

```diff
@@ -0,0 +1,143 @@
+# Automatic Testing
+
+In the `QED.jl` eco-system, we use [Continuous integration](https://en.wikipedia.org/wiki/Continuous_integration) (CIs) for running automatic tests. Each time, when a pull request is opened on GitHub.com or changes are committed to an existing pull request, the CI pipeline is triggered and starts a test script. The result of the tests will be reported in the pull request. In `QED.jl`, we distinguish between two kinds of tests
+
+- **Unit tests**: tests which check the functionality of the modified (sub) package. Those tests can either run standalone or use the functionality of other (third-party) packages.
+- **Integration tests**: tests which check the interoperatebility of the modified package with all sub-packages. For example, if package `A` depends on package `B` and you change package `B`, the integration test will check, if package `A` still works with the new version of package `B`.
+
+The CI will first execute the unit tests. Then, if all unit tests are passed, the integration tests will be started.
```

Comment:
Because unit tests test the functionality of the package itself. If the tests fails, it means that you package is broken and if you test if other packages can use your broken package, you will not expect, that it will work. Therefore it makes not sense to run integration tests, when the unit tests fails. It simply waste compute resources.

Reply:
Makes sense